### ISSUE TITLE
Launch minimized option doesn't work

### ISSUE THREAD

**Issue by iusmac** [telegramdesktop/tdesktop#26580]
### Steps to reproduce

Enable _Launch Telegram when system starts_ + _Launch minimized_ options

### Expected behaviour

```bash
$ grep Exec= ~/.config/autostart/org.telegram.desktop._4a4678b2ad6daffbfe9141935b9dd88b.desktop
Exec=/home/iusmac/.local/bin/telegram -autostart -startintray
```
_-startintray argument appended when enabling launch minimized option_

### Actual behaviour

```bash
$ grep Exec= ~/.config/autostart/org.telegram.desktop._4a4678b2ad6daffbfe9141935b9dd88b.desktop
Exec=/home/iusmac/.local/bin/telegram -autostart
```
_-startintray argument not appended when enabling launch minimized option_

### Operating system

Ubuntu 22.04.2 LTS

### Version of Telegram Desktop

4.8.9

### Installation source

Static binary from official website

**Comment by ilya-fedin** [comment]
`-startintray` is completely unrelated, it's about always starting in tray. `-autostart` is for autostart condition and reads autostart settings.

**Comment by iusmac** [comment]
> `-startintray` is completely unrelated, it's about always starting in tray (when launching Telegram from cmdline manually). `-autostart` is for autostart condition and reads autostart settings.

hm.. anyway, "Launch minimized" option doesn't work for me. I had to add `-startintray` manually.

**Comment by ilya-fedin** [comment]
maybe you have a passcode? The option won't work in that case while the `-startintray` forces.

**Comment by iusmac** [comment]
No, I don't have anything like this and have never used it before.

Here's my investigation.

Everything works on 4.8.4. If leaving TG untouched, it will automatically upgrade to 4.8.9 (latest) and nothing will break, the feature works as expected after re-login.

The issue appears immediately if manually upgrading from 4.8.4 to 4.8.5. And that's probably how I got this issue from last week's update.

Also, I can't incrementally upgrade from 4.8.5 to 4.8.6 and so on, as TG force upgrades itself to 4.8.9 (latest) on system boot, even if auto-update option is unchecked. So more user options totally ignored starting from 4.8.5 version.

In summary, the issue persists on all builds starting from 4.8.5 even if you do a fresh install of TG.

**Comment by ilya-fedin** [comment]
Have you tried to disable and re-enable the autostart option (not the launch minimized one)?

**Comment by Interpause** [comment]
I seem to have a similar issue, however for me, no permutation of `-autostart` and `-startintray` works to launch telegram minimized.

I am on version 4.8.9, Gentoo Linux (using binary package).

**Comment by ilya-fedin** [comment]
To launch minimized you need to enable the Launch minimized option in settings + use `-autostart`.

**Comment by ilya-fedin** [comment]
This may not work if you have no tray and your WM doesn't support minimization.

**Comment by Interpause** [comment]
I am using X on kwin 5.27.8, I do have a systray and telegram minimizes to it after I manually close the window.
